Automattic/newspack-popups
# feat: newsletter subscription list criteria

### All Submissions:






### Changes proposed in this Pull Request:

Implements two new criteria for segmentation:

- **Subscribed to newsletter lists**: matches a reader if they are subscribed to any of the selected lists
- **Not subscribed to newsletter lists**: matches a reader if they are not subscribed to any of the selected lists

For implementing these criteria, the `list__in` and `list__not_in` matching functions have been improved to also work for array values.

<img width="1096" alt="image" src="https://github.com/Automattic/newspack-popups/assets/820752/8697a5a2-3740-4e81-9bec-853b3de16d54">

### How to test the changes in this Pull Request:

1. Confirm written tests pass
2. Check out this branch and https://github.com/Automattic/newspack-plugin/pull/2658
3. Make sure you are connected to an ESP and have subscription lists enabled
4. Create a new campaign segment with "Subscribed to newsletter lists"
5. Add a prompt to this segment
6. In a fresh session, register a new reader account and subscribe to the list that matches the segment
7. Open the console and confirm you see the `np_reader_1_newsletter_subscribed_lists` item in `localStorage`
8. Confirm the prompt renders for the reader
9. Modify the segment to match readers **not** subscribed to the newsletter list
10. Confirm the prompt doesn't render for the reader
11. Start a fresh session and confirm the prompt renders for an anonymous reader

### Other information:
